src/Security/Authorization/Core/src/AuthorizationFailure.cs

```diff
@@ -39,6 +39,18 @@ public static AuthorizationFailure ExplicitFail()
                 FailCalled = true
             };
 
+        /// <summary>
+        /// Calls Fail and stores the failure reasons for future reference. Can be called multiple times. 
+        /// </summary>
+        /// <param name="reasons">The reasons of failure</param>
+        /// <returns>The failure.</returns>
+        public static AuthorizationFailure ExplicitFail(IEnumerable<AuthorizationFailureReason> reasons)
+            => new AuthorizationFailure
+            {
+                FailCalled = true,
+                FailureReasons = reasons
+            };
+
```

Comment:
You can remove this overload, it was outdated in the review. The new overload is the following one named `Failed`.